src/render_engine/collection.py

```diff
@@ -19,53 +19,71 @@
 
 class Collection(BaseObject):
     """
-    Collection objects serve as a way to quickly process pages that have a
-    portion of content that is similar or file driven.
+    Container for groups of related pages with shared properties and automatic features.
+
+    Collections provide a powerful way to manage multiple pages that share common
+    characteristics, such as blog posts, documentation pages, or product listings.
+    They automatically generate archives, feeds, and provide sorting/pagination.
```

Comment:
```suggestion
    They can generate archives, feeds, and provide sorting/pagination.
```

Reply:
This is important as these features are not enabled by default and should be set.